File: Java/110B-Lucky String.java
/*
	Solution to Codeforces problem 110B
	Copyright (c) GeneralMing. All rights reserved.

	https://github.com/GeneralMing/codeforces
*/

import java.util.*;
public class Lucky_String_110B
{
	public static void main(String[] args)
	{
		int n;
		Scanner in = new Scanner(System.in);
		n = Integer.parseInt(in.next());
		while(true)
		{
			if(n == 0)
				break;
			System.out.printf("a");
			n--;
			if(n == 0)
				break;
			System.out.printf("b");
			n--;
			if(n == 0)
				break;
			System.out.printf("c");
			n--;
			if(n == 0)
				break;
			System.out.printf("d");
			n--;
		}
		System.out.println("");
	}

}

File: Python/110B-Lucky String.py
"""
	Solution to Codeforces problem 110B
	Copyright (c) GeneralMing. All rights reserved.

	https://github.com/GeneralMing/codeforces
"""

n = int(input())
while(True):
	if(n == 0):
		break
	print("a", end = "")
	n -= 1
	if(n == 0):
		break
	print("b", end = "")
	n -= 1
	if(n == 0):
		break
	print("c", end = "")
	n -= 1
	if(n == 0):
		break
	print("d", end = "")
	n -= 1
print("")